src/decompiler.cc

```diff
@@ -540,13 +529,19 @@ struct Decompiler {
       auto is_import =
           CheckImportExport(ExternalKind::Func, func_index, f->name);
       AST ast(mc, f);
-      if (!is_import) ast.Construct(f->exprs, f->GetNumResults(), true);
+      LoadStoreTracking ls;
```

Comment:
This is weird. We take a pointer to a stack-allocated value, which has odd scopey lifetimey things.
Not actually sure what's ideal here. My gut would say have a class-scoped value, and then call a `reset` method for each function, but that requires explicitly clearing state and that seems actually-worse.
Maybe just null cur_ls out after each function?

Reply:
Given that the state of this thing is a single map, `reset` seems the way to go.